best-practice-on-public-cloud.md

```diff
@@ -0,0 +1,209 @@
+
+By using a dedicated 20GB [gp3](https://aws.amazon.com/ebs/general-purpose/) Raft Engine disk on AWS for a write-intensive social network application workload, the following improvements are observed but the estimated cost only increases by 0.4%:
+
+- a 17.5% increase in QPS (queries per second)
+- an 18.7% decrease in average latency for insert statements
+- a 45.6% decrease in p99 latency for insert statements.
+
+
+| Metric | Shared Raft Engine disk | Dedicated Raft Engine disk | Difference (%) |
+| ------------- | ------------- |------------- |------------- |
+| QPS (K/s)| 8.0 | 9.4 | 17.5| 
+| AVG Insert Latency (ms)| 11.3 | 9.2 | -18.7 |
+| P99 Insert Latency (ms)| 29.4 | 16.0 | -45.6|
+
+### Example 2: Run TPC-C/SYSBench workload on Azure
+
+By using a dedicated 32GB [ultra disk](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disks) for Raft Engine on Azure, the following improvements are observed:
+
+- Sysbench `oltp_read_write` workload: a 17.8% increase in QPS and a 15.6% decrease in average latency.
+- TPC-C workload: a 27.6% increase in QPS and a 23.1% decrease in average latency.
+
+| Metric | Workload | Shared Raft Engine disk | Dedicated Raft Engine disk | Difference (%) |
+| ------------- | ------------- | ------------- |------------- |------------- |
+| QPS (K/s) | Sysbench `oltp_read_write` | 60.7 | 71.5 | 17.8| 
+| QPS (K/s) | TPC-C | 23.9 | 30.5 | 27.6| 
+| AVG Latency (ms)| Sysbench `oltp_read_write` |  4.5 | 3.8 | -15.6 |
+| AVG Latency (ms)| TPC-C |  3.9 | 3.0 | -23.1 |
+
+### Example 3: Attach a dedicated pd-ssd disk on Google Cloud for Raft Engine on TiKV manifest
+
+The following TiKV configuration example shows how to attach an additional 512GB [pd-ssd](https://cloud.google.com/compute/docs/disks#disk-types/) disk to a cluster on Google Cloud deployed by [TiDB Operator](https://docs.pingcap.com/tidb-in-kubernetes/stable), with `raft-engine.dir` configured to store Raft Engine logs to this specific disk.
+
+```
+tikv:
+    config: |
+      [raft-engine]
+        dir = "/var/lib/raft-pv-ssd/raft-engine"
+        enable = true
+        enable-log-recycle = true
+    requests:
+      storage: 4Ti
+    storageClassName: pd-ssd
+    storageVolumes:
+    - mountPath: /var/lib/raft-pv-ssd
+      name: raft-pv-ssd
+      storageSize: 512Gi
+```
+
+
+## Reduce compaction IO flow in KV RocksDB
+
+As the storage engine of TiKV, [RocksDB](https://rocksdb.org/) is used to store user data. However, because the provisioned IO throughput on cloud EBS is usually limited due to cost considerations, RocksDB might exhibit high write amplification, and the disk throughput might become the bottleneck for the workload. As a result, the total number of pending compaction bytes grows over time and triggers flow control, which indicates that TiKV lacks sufficient disk bandwidth to keep up with the foreground write flow. 
```

Comment:
```suggestion
As the storage engine of TiKV, [RocksDB](https://rocksdb.org/) is used to store user data. Because the provisioned IO throughput on cloud EBS is usually limited due to cost considerations, RocksDB might exhibit high write amplification, and the disk throughput might become the bottleneck for the workload. As a result, the total number of pending compaction bytes grows over time and triggers flow control, which indicates that TiKV lacks sufficient disk bandwidth to keep up with the foreground write flow. 
```